python/ray/includes/gcs_client.pxi

```diff
@@ -534,11 +534,15 @@ cdef convert_get_all_node_info(
     try:
         check_status_timeout_as_rpc_error(status)
         node_table_data = {}
-        for c_proto in c_data:
-            b = c_proto.SerializeAsString()
-            proto = gcs_pb2.GcsNodeInfo()
-            proto.ParseFromString(b)
-            node_table_data[NodeID.from_binary(proto.node_id)] = proto
+        # No GIL for C++ looping && serialization.
+        # GIL for Pyhton deserialization and dict building.
+        with nogil:
+            for c_proto in c_data:
+                b = c_proto.SerializeAsString()
```

Comment:
For now, let's do the `nogil` work together and then `gil` work to match the existing behavior just to be safe.